### DISCUSSION TITLE
Customize CdkPipeline source action

### DISCUSSION THREAD

**Discussion by asterikx** [aws/aws-cdk#16553] [Q&A]
How can I customize the CodeCommit source action of a `CodePipeline` to use `Git clone depth: Default`? In the console the created source action shows `Git clone depth: Full` (official docs on [default vs full clone](https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-CodeCommit.html)).
The cloned repo is rather large (1,5 GB) and the source action takes ~5 minutes. I hope to speed up the source action by shallow cloning instead of full cloning.

**Comment by peterwoodworth** [comment]
It's not clear exactly which of our APIs you are referring to or using. But our pipelines library has received significant updates since this discussion has submitted, please view the new API https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.pipelines-readme.html
